File: Daily/PY/Leetcode645-出错的集合.py
from typing import List
class Solution:
    def findErrorNums(self, nums: List[int]) -> List[int]:
        n = len(nums)
        ans = []
        dt = [0] * (len(nums) + 1)
        for num in nums:
            dt[num] += 1
            if dt[num] > 1:
                ans.append(num)
        
        for i in range(1,n + 1):
            if dt[i] == 0:
                ans.append(i)
                break

        return ans

File: Leetcode/sbgo/Leetcode645-出错的集合.go
package sbgo

func findErrorNums(nums []int) []int {
	n := len(nums)
	dt := make([]int, n+1)
	ans := []int{}
	for _, num := range nums {
		dt[num]++
		if dt[num] > 1 {
			ans = append(ans, num)
		}
	}
	for i := 1; i <= n; i++ {
		if dt[i] == 0 {
			ans = append(ans, i)
			break
		}
	}
	return ans
}
